envi/cli.py

```diff
@@ -476,6 +478,17 @@ def do_script(self, line):
         locals = self.getExpressionLocals()
         locals['argv'] = argv
 
+        if len(argv) and argv[0] == "?":
+            scripts = []
+            for scriptdir in self.scriptpaths:
+                # FIXME: filter on more than just ".py".  something internal
+                pscripts = [py for py in os.listdir(scriptdir) if self.validate_script(scriptdir + os.sep + py)]
```

Comment:
Couple more things, the FIXME above seems to imply that there is "*.py" name filtering going on but i dont see any...

Also, the list comprehension here is not perf oriented ( i try to avoid them for clarity except in speed cases mostly ) and if broken out would allow some better checks like:
```
pypath = os.path.join(scriptdir,py)
if not os.path.isfile(pypath):
    continue
```
rather than using exception handling as part of intended program logic ( ie, where dir names are knowingly handed in to the validate_script() method and purposely exceptioning ) which should be avoided if possible.